dstep/translator/Type.d

```diff
@@ -383,6 +386,21 @@ body
     }
 }
 
+private Token tokenInsideSquareBrackets(Cursor cursor)
+{
+    import std.algorithm: find;
+    import clang.Token: TokenKind;
+
+    // find the first token after the first opening square bracket
+    auto tokenRange = cursor.tokens
+        .find!(a => a.kind == TokenKind.punctuation && a.spelling == "[");
+    if(tokenRange.empty) return Token.init;
```

Comment:
If the range is empty there's no way to tell that from the return value. Unfortunately it seems it's not possible to currently have an invalid `Token`. Perhaps we should add a new token kind, `invalid`.